src/solvers/floatbv/float_bv.cpp

```diff
@@ -20,76 +20,145 @@ exprt float_bvt::convert(const exprt &expr) const
+    return not_exprt(is_equal(
+      notequal_expr.lhs(), notequal_expr.rhs(), get_spec(notequal_expr.lhs())));
+  }
   else if(expr.id()==ID_floatbv_typecast)
   {
-    const typet &src_type=expr.op0().type();
-    const typet &dest_type=expr.type();
+    const auto &floatbv_typecast_expr = to_floatbv_typecast_expr(expr);
+    const auto &op = floatbv_typecast_expr.op();
+    const typet &src_type = floatbv_typecast_expr.op().type();
+    const typet &dest_type = floatbv_typecast_expr.type();
+    const auto &rounding_mode = floatbv_typecast_expr.rounding_mode();
 
     if(dest_type.id()==ID_signedbv &&
        src_type.id()==ID_floatbv) // float -> signed
-      return
-        to_signed_integer(
-          expr.op0(),
-          to_signedbv_type(dest_type).get_width(),
-          expr.op1(),
-          get_spec(expr.op0()));
+      return to_signed_integer(
+        op,
+        to_signedbv_type(dest_type).get_width(),
+        rounding_mode,
+        get_spec(op));
     else if(dest_type.id()==ID_unsignedbv &&
             src_type.id()==ID_floatbv) // float -> unsigned
-      return
-        to_unsigned_integer(
-          expr.op0(),
-          to_unsignedbv_type(dest_type).get_width(),
-          expr.op1(),
-          get_spec(expr.op0()));
+      return to_unsigned_integer(
+        op,
+        to_unsignedbv_type(dest_type).get_width(),
+        rounding_mode,
+        get_spec(op));
     else if(src_type.id()==ID_signedbv &&
             dest_type.id()==ID_floatbv) // signed -> float
-      return from_signed_integer(
-        expr.op0(), expr.op1(), get_spec(expr));
+      return from_signed_integer(op, rounding_mode, get_spec(expr));
     else if(src_type.id()==ID_unsignedbv &&
             dest_type.id()==ID_floatbv) // unsigned -> float
-      return from_unsigned_integer(
-        expr.op0(), expr.op1(), get_spec(expr));
+    {
+      return from_unsigned_integer(op, rounding_mode, get_spec(expr));
+    }
     else if(dest_type.id()==ID_floatbv &&
             src_type.id()==ID_floatbv) // float -> float
-      return
-        conversion(
-          expr.op0(), expr.op1(), get_spec(expr.op0()), get_spec(expr));
+    {
+      return conversion(op, rounding_mode, get_spec(op), get_spec(expr));
+    }
     else
       return nil_exprt();
   }
-  else if(expr.id()==ID_typecast &&
-          expr.type().id()==ID_bool &&
-          expr.op0().type().id()==ID_floatbv)  // float -> bool
-    return not_exprt(is_zero(expr.op0()));
+  else if(
+    expr.id() == ID_typecast && expr.type().id() == ID_bool &&
+    to_typecast_expr(expr).op().type().id() == ID_floatbv) // float -> bool
+  {
+    return not_exprt(is_zero(to_typecast_expr(expr).op()));
+  }
   else if(expr.id()==ID_floatbv_plus)
-    return add_sub(false, expr.op0(), expr.op1(), expr.op2(), get_spec(expr));
+  {
+    const auto &float_expr = to_ieee_float_op_expr(expr);
+    return add_sub(
+      false,
+      float_expr.op0(),
+      float_expr.op1(),
```

Comment:
`lhs()` and `rhs()` for consistency with using `rounding_mode()` not `op2()`.

Reply:
done